apps/server/src/shared/repo/task/task.repo.ts

```diff
@@ -1,14 +1,77 @@
 import { Injectable } from '@nestjs/common';
-import { EntityManager, FilterQuery, QueryOrderMap } from '@mikro-orm/core';
+import { FilterQuery, QueryOrderMap } from '@mikro-orm/core';
+import { EntityManager } from '@mikro-orm/mongodb';
 
-import { EntityId, IFindOptions, Task, Counted } from '@shared/domain';
+import { EntityId, IFindOptions, Task, Counted, SortOrder } from '@shared/domain';
 
 import { TaskScope } from './task-scope';
 
 @Injectable()
 export class TaskRepo {
 	constructor(private readonly em: EntityManager) {}
 
+	async findAllFinishedByParentIds(
+		parentIds: {
+			creatorId: EntityId;
+			openCourseIds: EntityId[];
+			lessonIdsOfOpenCourses: EntityId[];
+			finishedCourseIds: EntityId[];
+			lessonIdsOfFinishedCourses: EntityId[];
+		},
+		options?: IFindOptions<Task>
+	): Promise<Counted<Task[]>> {
+		const { pagination } = options || {};
+
+		const scope = new TaskScope('$or');
+
+		const parentsOpen = new TaskScope('$or');
+		parentsOpen.byCourseIds(parentIds.openCourseIds);
+		parentsOpen.byLessonIds(parentIds.lessonIdsOfOpenCourses);
+
+		const parentsFinished = new TaskScope('$or');
+		parentsFinished.byCourseIds(parentIds.finishedCourseIds);
+		parentsFinished.byLessonIds(parentIds.lessonIdsOfFinishedCourses);
+
+		const openScope = new TaskScope();
```

Comment:
we should maybe rename the scope. how about `finishedByTeacherWithOpenCourseOrLessonScope`? it's a little bit long but describes better what this is about.